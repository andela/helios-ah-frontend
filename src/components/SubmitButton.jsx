import React from 'react';
import PropTypes from 'prop-types';

const SubmitButton = ({
  onClick,
  isRequestSent,
  value,
  id,
  className,
  columnAttribute
}) => (
  <div className={columnAttribute}>
    <div className="form-group text-center">
      <button
        type="submit"
        className={className}
        id={id}
        onClick={onClick}
        disabled={isRequestSent}
      >
        {value}
      </button>
    </div>
  </div>
);

SubmitButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  isRequestSent: PropTypes.bool,
  value: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  columnAttribute: PropTypes.string,
};

SubmitButton.defaultProps = {
  isRequestSent: false,
  columnAttribute: ''
};

export default SubmitButton;
