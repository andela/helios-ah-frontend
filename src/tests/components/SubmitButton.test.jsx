import React from 'react';
import { shallow } from 'enzyme';
import chai from 'chai';
import SubmitButton from '../../components/SubmitButton';

const { expect } = chai;
describe('Unit test for the submit button component. ', () => {
  it('should render an html with a button tag', () => {
    const propObj = {
      submitDetails: () => {},
      isRequestSent: true,
      buttonValue: 'buttonValue',
      idAttribute: 'login-button',
      buttonClass: 'someClassProperty',
      columnAttribute: 'col-md-4'
    };
    const wrapper = shallow(<SubmitButton {...propObj} />);
    expect(wrapper.find('button').text()).to.equal('buttonValue');
    expect(wrapper.find('button').length).to.equal(1);
    expect(wrapper.find('button').is('#login-button')).to.equal(true);
    expect(wrapper.find('button').is('.someClassProperty')).to.equal(true);
    expect(wrapper.is('.col-md-4')).to.equal(true);
  });
});
