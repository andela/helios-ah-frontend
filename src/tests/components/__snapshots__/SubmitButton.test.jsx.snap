// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the submit button component.  Render Submit button it should render submit button 1`] = `
<div
  className="signup-col"
>
  <div
    className="form-group text-center"
  >
    <button
      className="btn"
      disabled={true}
      id={null}
      onClick={[Function]}
      type="submit"
    >
      SIGNUP
    </button>
  </div>
</div>
`;
